### PR TITLE
colexec: optimize aggregate functions

### PR DESCRIPTION
**Pull request by yuzefovich**
**colexec: simplify aggregateFunc interface a bit**

This commit simplifies `SetOutputIndex` to no longer be required to
unset nulls after the desired output index. The unsetting is only
necessary when we have "overflow" tuples in the ordered aggregator, so
it makes that it should be the one resetting the nulls.

Additionally, the implementation of `SetOutputIndex` no longer checks
whether `curIdx` is not -1 (meaning the function has already performed
some aggregation). I think this was an artifact from the past and is no
longer necessary.

This commit also makes a minor performance optimization to unwrap the
output batch in the ordered aggregator.

Release note: None

**colexec: optimize any_not_null func for hash aggregation**

When we're performing hash aggregation, we know that there will be
exactly one group. This insight allows us optimize the aggregate
functions, and this commit performs the optimization of any_not_null.
Now we will have two separate files for both aggregator kinds (hash and
ordered), and that part is handled by a preprocessing step that replaces
`_AGGKIND` template "variable" with the corresponding aggregator kind.

I looked into introducing some base struct for aggregates to reuse, but
it is rather complicated at the moment - some aggregate functions
iterate over `[]*oneArgOverload`, and we could flatten that out.
However, other functions operate with a custom "tmpl info" struct.
Furthermore, extending the struct to include `AggKind` variable would
require refactoring the templates themselves because we assume that `.`
(dot) of the template usually is at `lastArgWidthOverload` struct.
Probably in order to provide some "aggregate base tmpl info" struct that
would help with hash vs ordered aggregation generation we would need to
introduce an interface which custom tmpl infos will implement.

Release note: None

**colexec: optimize all aggregate funcs for hash**

This commit adds a simple optimization of omitting the check for a start
of a new group in all aggregate functions if we're performing hash
aggregation.

This change also allows us to slightly simplify the hash aggregator.

Release note: None

**colexec: optimize aggregateFunc contract a bit**

This commit adjusts the contract of `aggregateFunc.Init` method to
require that the very first tuple in the whole input should *not* have
a `true` value set in `groups` vector. This requirement allows us to
optimize all aggregate functions for the case of ordered aggregator. The
ordered aggregator has been adjusted accordingly to satisfy the
contract.

This commit also marks `oneShotOp` as `NonExplainable`.

Release note: None

### CODE REVIEW

**Comment by yuzefovich** [comment]
The benchmarks are [here](https://gist.github.com/yuzefovich/94e9699a867a4ef7a3e8c10d66cdef1a). This commit slightly improves the speed of hash aggregation at the cost of decreasing the speed of ordered aggregation, and I think this tradeoff is acceptable given the absolute speed of ordered is still very high.

Optimization of any_not_null deserves a call out:
```
Aggregator/ANY_NOT_NULL/hash/int/groupSize=1/hasNulls=false/numInputBatches=64-24            5.19MB/s ± 3%   5.42MB/s ± 2%   +4.51%  (p=0.000 n=9+10)
Aggregator/ANY_NOT_NULL/hash/int/groupSize=1/hasNulls=true/numInputBatches=64-24             4.92MB/s ± 3%   5.40MB/s ± 2%   +9.65%  (p=0.000 n=10+10)
Aggregator/ANY_NOT_NULL/hash/int/groupSize=2/hasNulls=false/numInputBatches=64-24            10.9MB/s ± 2%   11.1MB/s ± 2%   +1.90%  (p=0.001 n=10+9)
Aggregator/ANY_NOT_NULL/hash/int/groupSize=2/hasNulls=true/numInputBatches=64-24             10.1MB/s ± 1%   11.2MB/s ± 1%  +11.48%  (p=0.000 n=10+10)
Aggregator/ANY_NOT_NULL/hash/int/groupSize=512/hasNulls=false/numInputBatches=64-24           464MB/s ± 1%    534MB/s ± 0%  +14.99%  (p=0.000 n=10+10)
Aggregator/ANY_NOT_NULL/hash/int/groupSize=512/hasNulls=true/numInputBatches=64-24            460MB/s ± 1%    532MB/s ± 1%  +15.52%  (p=0.000 n=9+9)
Aggregator/ANY_NOT_NULL/hash/int/groupSize=1024/hasNulls=false/numInputBatches=64-24          496MB/s ± 0%    566MB/s ± 0%  +14.10%  (p=0.000 n=9+10)
Aggregator/ANY_NOT_NULL/hash/int/groupSize=1024/hasNulls=true/numInputBatches=64-24           490MB/s ± 0%    564MB/s ± 0%  +15.15%  (p=0.000 n=10+10)
Aggregator/ANY_NOT_NULL/hash/decimal/groupSize=1/hasNulls=false/numInputBatches=64-24        4.65MB/s ± 3%   4.71MB/s ± 3%     ~     (p=0.269 n=10+10)
Aggregator/ANY_NOT_NULL/hash/decimal/groupSize=1/hasNulls=true/numInputBatches=64-24         4.44MB/s ± 1%   4.84MB/s ± 2%   +8.96%  (p=0.000 n=9+9)
Aggregator/ANY_NOT_NULL/hash/decimal/groupSize=2/hasNulls=false/numInputBatches=64-24        9.63MB/s ± 2%   9.44MB/s ±11%     ~     (p=0.781 n=10+10)
Aggregator/ANY_NOT_NULL/hash/decimal/groupSize=2/hasNulls=true/numInputBatches=64-24         8.69MB/s ± 9%   9.90MB/s ± 1%  +14.02%  (p=0.000 n=10+10)
Aggregator/ANY_NOT_NULL/hash/decimal/groupSize=512/hasNulls=false/numInputBatches=64-24       440MB/s ± 2%    529MB/s ± 1%  +20.25%  (p=0.000 n=9+10)
Aggregator/ANY_NOT_NULL/hash/decimal/groupSize=512/hasNulls=true/numInputBatches=64-24        440MB/s ± 3%    528MB/s ± 1%  +19.82%  (p=0.000 n=10+9)
Aggregator/ANY_NOT_NULL/hash/decimal/groupSize=1024/hasNulls=false/numInputBatches=64-24      470MB/s ± 1%    565MB/s ± 1%  +20.36%  (p=0.000 n=10+10)
Aggregator/ANY_NOT_NULL/hash/decimal/groupSize=1024/hasNulls=true/numInputBatches=64-24       476MB/s ± 1%    564MB/s ± 1%  +18.37%  (p=0.000 n=10+10)
Aggregator/ANY_NOT_NULL/hash/bytes/groupSize=1/hasNulls=false/numInputBatches=64-24          4.86MB/s ± 3%   4.98MB/s ± 2%   +2.45%  (p=0.004 n=10+10)
Aggregator/ANY_NOT_NULL/hash/bytes/groupSize=1/hasNulls=true/numInputBatches=64-24           4.64MB/s ± 1%   4.98MB/s ± 0%   +7.36%  (p=0.000 n=9+8)
Aggregator/ANY_NOT_NULL/hash/bytes/groupSize=2/hasNulls=false/numInputBatches=64-24          10.1MB/s ± 3%   10.5MB/s ± 1%   +3.75%  (p=0.000 n=10+9)
Aggregator/ANY_NOT_NULL/hash/bytes/groupSize=2/hasNulls=true/numInputBatches=64-24           9.43MB/s ± 2%  10.37MB/s ± 4%  +10.01%  (p=0.000 n=10+10)
Aggregator/ANY_NOT_NULL/hash/bytes/groupSize=512/hasNulls=false/numInputBatches=64-24         423MB/s ± 1%    460MB/s ± 4%   +8.68%  (p=0.000 n=10+10)
Aggregator/ANY_NOT_NULL/hash/bytes/groupSize=512/hasNulls=true/numInputBatches=64-24          398MB/s ± 1%    461MB/s ± 2%  +15.61%  (p=0.000 n=10+10)
Aggregator/ANY_NOT_NULL/hash/bytes/groupSize=1024/hasNulls=false/numInputBatches=64-24        448MB/s ± 0%    502MB/s ± 1%  +11.99%  (p=0.000 n=10+10)
Aggregator/ANY_NOT_NULL/hash/bytes/groupSize=1024/hasNulls=true/numInputBatches=64-24         421MB/s ± 1%    503MB/s ± 1%  +19.60%  (p=0.000 n=10+10)
```

**Review by asubiotto — CHANGES_REQUESTED** [review]
> Reviewed 8 of 8 files at r1, 7 of 7 files at r2, 13 of 13 files at r3.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/49934)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @yuzefovich)

___
*[Makefile, line 834 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/49934#-M9Yyzun4i3cSsiRUpNd:-M9Yyzun4i3cSsiRUpNe:bydiu7) ([raw file](https://github.com/cockroachdb/cockroach/blob/d0e11807cd5cee33af80b51571cb9db99b3d2307/Makefile#L834)):*
> ```Makefile
>   pkg/sql/colexec/and_or_projection.eg.go \
>   pkg/sql/colexec/any_not_null_hash_agg.eg.go \
>   pkg/sql/colexec/any_not_null_ordered_agg.eg.go \
> ```

Is there going to be a problem if we double the number of files? To be clear, I prefer having these in separate files.
___
*[pkg/sql/colexec/any_not_null_agg_tmpl.go, line 107 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/49934#-M9YwRco7F0kijsyBngM:-M9YwRcpBGYbyRWTT4iZ:bl330g5) ([raw file](https://github.com/cockroachdb/cockroach/blob/432a1c53bdc486fe4474d45df045b9fd87ebfc53/pkg/sql/colexec/any_not_null_agg_tmpl.go#L107)):*
> ```Go
> 
> func (a *anyNotNull_TYPEAgg) SetOutputIndex(idx int) {
> 	if a.curIdx != -1 {
> ```

Can you expand on why this is no longer necessary?
___
*[pkg/sql/colexec/any_not_null_agg_tmpl.go, line 91 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/49934#-M9YyoKm8Uap8_bRdZoU:-M9YyoKm8Uap8_bRdZoV:b4tx7ze) ([raw file](https://github.com/cockroachdb/cockroach/blob/d0e11807cd5cee33af80b51571cb9db99b3d2307/pkg/sql/colexec/any_not_null_agg_tmpl.go#L91)):*
> ```Go
> 
> func (a *anyNotNull_TYPE_AGG_KINDAgg) Init(groups []bool, vec coldata.Vec) {
> 	// _IF_CAN_HAVE_MULTIPLE_GROUPS
> ```

Similar to the other PR, I think this is another case where maybe we could explore extracting this logic into a `anyNotNullTmplInfo` struct. I realize this could be overkill for this PR but I think we should think about how we could make these template info structs composable. i.e. in an ideal world we'd define a "base" any not null aggregation (possibly even a "base" aggregation template info struct that would finally get rid of rewriting the same logic in all aggregates) that would somehow be reused in a "custom " any not null aggregation struct that would include groups. I'm sort of thinking out loud so I don't know if this makes sense or if it'll work, but I think a big problem is that we can't compose templates. It's kind of like the same idea as the refactor I did of the tpchvec tests. There was a lot of "if we're running this test do this, otherwise do that", that became a lot simpler when composing them.
___
*[pkg/sql/colexec/ordered_aggregator.go, line 59 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/49934#-M9Yw_5P8n_N2DE6P-D3:-M9Yw_5P8n_N2DE6P-D4:by2ppou) ([raw file](https://github.com/cockroachdb/cockroach/blob/16d5521b7cad659c977bf55a56bd600c0012122d/pkg/sql/colexec/ordered_aggregator.go#L59)):*
> ```Go
> 	scratch struct {
> 		coldata.Batch
> 		vecs  []coldata.Vec
> ```

What is the concrete performance improvement we get from this change?
___
*[pkg/sql/colexec/execgen/cmd/execgen/avg_agg_gen.go, line 105 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/49934#-M9Z4yPFEomCTXrZWYQy:-M9Z4yPFEomCTXrZWYQz:b-s6h4rg) ([raw file](https://github.com/cockroachdb/cockroach/blob/534ae4c691e6428411f7fa921743df9cb4c812bb/pkg/sql/colexec/execgen/cmd/execgen/avg_agg_gen.go#L105)):*
> ```Go
> 
> func init() {
> 	registerAggGenerator(genAvgAgg, "avg_%s_agg.eg.go", avgAggTmpl)
> ```

Should we prepend the "hash"/"ordered" string so that we can avoid a formatting directive in this string? It seems out of place.

**Review by yuzefovich — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/49934)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @asubiotto)

___
*[Makefile, line 834 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/49934#-M9Yyzun4i3cSsiRUpNd:-M9aifcO5OWxpYCuuPbu:bpokfx7) ([raw file](https://github.com/cockroachdb/cockroach/blob/d0e11807cd5cee33af80b51571cb9db99b3d2307/Makefile#L834)):*
<details><summary><i>Previously, asubiotto (Alfonso Subiotto Marqués) wrote…</i></summary><blockquote>

Is there going to be a problem if we double the number of files? To be clear, I prefer having these in separate files.
</blockquote></details>

The only "problem" I can see is that `Makefile` is getting larger and there will be more entries in the build log related to execgen. It shouldn't have influence on the performance of CockroachDB nor on the build time.
___
*[pkg/sql/colexec/any_not_null_agg_tmpl.go, line 107 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/49934#-M9YwRco7F0kijsyBngM:-M9al3gC6n_muMwdOUH4:b4behnr) ([raw file](https://github.com/cockroachdb/cockroach/blob/432a1c53bdc486fe4474d45df045b9fd87ebfc53/pkg/sql/colexec/any_not_null_agg_tmpl.go#L107)):*
<details><summary><i>Previously, asubiotto (Alfonso Subiotto Marqués) wrote…</i></summary><blockquote>

Can you expand on why this is no longer necessary?
</blockquote></details>

I have a returning question - why was it ever necessary? My guess is that initially there was a different handling of groups in the ordered aggregator, but now we call `SetOutputIndex` only when we have accumulated more groups that can fit into the output batch, so it is now guaranteed that `curIdx` is not `-1`.
___
*[pkg/sql/colexec/any_not_null_agg_tmpl.go, line 91 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/49934#-M9YyoKm8Uap8_bRdZoU:-M9anI720k8TpVYquPs6:b-xy8qxs) ([raw file](https://github.com/cockroachdb/cockroach/blob/d0e11807cd5cee33af80b51571cb9db99b3d2307/pkg/sql/colexec/any_not_null_agg_tmpl.go#L91)):*
<details><summary><i>Previously, asubiotto (Alfonso Subiotto Marqués) wrote…</i></summary><blockquote>

Similar to the other PR, I think this is another case where maybe we could explore extracting this logic into a `anyNotNullTmplInfo` struct. I realize this could be overkill for this PR but I think we should think about how we could make these template info structs composable. i.e. in an ideal world we'd define a "base" any not null aggregation (possibly even a "base" aggregation template info struct that would finally get rid of rewriting the same logic in all aggregates) that would somehow be reused in a "custom " any not null aggregation struct that would include groups. I'm sort of thinking out loud so I don't know if this makes sense or if it'll work, but I think a big problem is that we can't compose templates. It's kind of like the same idea as the refactor I did of the tpchvec tests. There was a lot of "if we're running this test do this, otherwise do that", that became a lot simpler when composing them.
</blockquote></details>

I see your point, and I agree that it would be nice to have composable structs or a "base" any not null (or aggregate in general) struct.

However, I want to be clear on why I opened up this PR: I realized that there could be an easy gain for `any_not_null` aggregate in particular but also, because I intend on adding the support of filtering aggregation, and in order to support that for ordered aggregator I think I will need to plumb the filter directly into aggregate function itself, but the hash aggregator doesn't need that. So I decided to take a step and customize two aggregate kinds using the existing infra we have. Improving the templates in general wasn't my goal, and it should be planned for and worked on separately.

I think this PR doesn't make the existing situation much worse by introducing some custom text templating logic, and to me `_IF_CAN_HAVE_MULTIPLE_GROUPS` is very explicit, so there should be no confusion on what it means in terms of readability of templates. Additionally, this logic is shared among all aggregate functions, and I'm thinking it should be easy to update once we decide on what we want to do with the templating infra. So I'm inclined to keep my original approach unless you have a strong objection.
___
*[pkg/sql/colexec/ordered_aggregator.go, line 59 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/49934#-M9Yw_5P8n_N2DE6P-D3:-M9auwZe5_x5GtWQqwjW:bb6mw1h) ([raw file](https://github.com/cockroachdb/cockroach/blob/16d5521b7cad659c977bf55a56bd600c0012122d/pkg/sql/colexec/ordered_aggregator.go#L59)):*
<details><summary><i>Previously, asubiotto (Alfonso Subiotto Marqués) wrote…</i></summary><blockquote>

What is the concrete performance improvement we get from this change?
</blockquote></details>

The [benchmarks](https://gist.github.com/yuzefovich/895cde900c95532bd97c461d5507fe51) show very negligible performance difference (mostly gains, but I expected a larger difference tbh), so I'm neutral towards keeping this unwrapping of the scratch batch. If you think it makes the code worse, I'll remove it.
___
*[pkg/sql/colexec/execgen/cmd/execgen/avg_agg_gen.go, line 105 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/49934#-M9Z4yPFEomCTXrZWYQy:-M9ampI96ya-StDeCERx:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/534ae4c691e6428411f7fa921743df9cb4c812bb/pkg/sql/colexec/execgen/cmd/execgen/avg_agg_gen.go#L105)):*
<details><summary><i>Previously, asubiotto (Alfonso Subiotto Marqués) wrote…</i></summary><blockquote>

Should we prepend the "hash"/"ordered" string so that we can avoid a formatting directive in this string? It seems out of place.
</blockquote></details>

Done.

**Review by yuzefovich — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/49934)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @asubiotto)

___
*[pkg/sql/colexec/any_not_null_agg_tmpl.go, line 91 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/49934#-M9YyoKm8Uap8_bRdZoU:-M9dod1Z7MVfMketQZrY:b6xlpq4) ([raw file](https://github.com/cockroachdb/cockroach/blob/d0e11807cd5cee33af80b51571cb9db99b3d2307/pkg/sql/colexec/any_not_null_agg_tmpl.go#L91)):*
<details><summary><i>Previously, yuzefovich wrote…</i></summary><blockquote>

I see your point, and I agree that it would be nice to have composable structs or a "base" any not null (or aggregate in general) struct.

However, I want to be clear on why I opened up this PR: I realized that there could be an easy gain for `any_not_null` aggregate in particular but also, because I intend on adding the support of filtering aggregation, and in order to support that for ordered aggregator I think I will need to plumb the filter directly into aggregate function itself, but the hash aggregator doesn't need that. So I decided to take a step and customize two aggregate kinds using the existing infra we have. Improving the templates in general wasn't my goal, and it should be planned for and worked on separately.

I think this PR doesn't make the existing situation much worse by introducing some custom text templating logic, and to me `_IF_CAN_HAVE_MULTIPLE_GROUPS` is very explicit, so there should be no confusion on what it means in terms of readability of templates. Additionally, this logic is shared among all aggregate functions, and I'm thinking it should be easy to update once we decide on what we want to do with the templating infra. So I'm inclined to keep my original approach unless you have a strong objection.
</blockquote></details>

I looked into this for a little bit, and here is what I put in the commit message:
```
I looked into introducing some base struct for aggregates to reuse, but
it is rather complicated at the moment - some aggregate functions
iterate over `[]*oneArgOverload`, and we could flatten that out.
However, other functions operate with a custom "tmpl info" struct.
Furthermore, extending the struct to include `AggKind` variable would
require refactoring the templates themselves because we assume that `.`
(dot) of the template usually is at `lastArgWidthOverload` struct.
Probably in order to provide some "aggregate base tmpl info" struct that
would help with hash vs ordered aggregation generation we would need to
introduce an interface which custom tmpl infos will implement.
```
TL;DR is that it is rather complicated to attempt to introduce some composability of struct for the template generation.

**Comment by yuzefovich** [comment]
I added fourth commit to this PR that adjusts `aggregateFunc` contract which gives us nice performance boosts in some cases for ordered aggregator. The benchmarks of the fourth commit are [here](https://gist.github.com/yuzefovich/87bd0fea27e48fb743cf825fd834326e).

**Comment by asubiotto** [comment]
Did you mean to rebase on top of the other branch? I'll take a look at the new changes once that's merged.

**Comment by yuzefovich** [comment]
> Did you mean to rebase on top of the other branch?

Yeah, I resolved the merge conflicts and wanted to make sure that nothing breaks.

> I'll take a look at the new changes once that's merged.

Sounds good.

**Comment by yuzefovich** [comment]
Rebased, RFAL.

**Review by asubiotto — APPROVED** [review]
:lgtm: 

> Reviewed 17 of 17 files at r4, 7 of 7 files at r5, 13 of 13 files at r6, 11 of 11 files at r7.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/49934#-:-M9xRq390VrZubwUEXuF:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @asubiotto and @yuzefovich)

___
*[pkg/sql/colexec/any_not_null_agg_tmpl.go, line 107 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/49934#-M9YwRco7F0kijsyBngM:-M9xP7h392q78j7sF3BF:b-q22qrx) ([raw file](https://github.com/cockroachdb/cockroach/blob/432a1c53bdc486fe4474d45df045b9fd87ebfc53/pkg/sql/colexec/any_not_null_agg_tmpl.go#L107)):*
<details><summary><i>Previously, yuzefovich wrote…</i></summary><blockquote>

I have a returning question - why was it ever necessary? My guess is that initially there was a different handling of groups in the ordered aggregator, but now we call `SetOutputIndex` only when we have accumulated more groups that can fit into the output batch, so it is now guaranteed that `curIdx` is not `-1`.
</blockquote></details>

:shrug: if it works. Yeah, I think the previous code called `SetOutputIndex` before aggregation computation.
___
*[pkg/sql/colexec/ordered_aggregator.go, line 59 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/49934#-M9Yw_5P8n_N2DE6P-D3:-M9xPIgj5UryKl56dHWJ:b-1718y) ([raw file](https://github.com/cockroachdb/cockroach/blob/16d5521b7cad659c977bf55a56bd600c0012122d/pkg/sql/colexec/ordered_aggregator.go#L59)):*
<details><summary><i>Previously, yuzefovich wrote…</i></summary><blockquote>

The [benchmarks](https://gist.github.com/yuzefovich/895cde900c95532bd97c461d5507fe51) how very negligible performance difference (mostly gains, but I expected a larger difference tbh), so I'm neutral towards keeping this unwrapping of the scratch batch. If you think it makes the code worse, I'll remove it.
</blockquote></details>

I think it does because there are now two ways to do the same thing. I would remove it.

**Review by yuzefovich — COMMENTED** [review]
TFTR!

bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/49934#-:-M9xaXrEDNU09NHBzmnk:b-oduykh)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @asubiotto)

___
*[pkg/sql/colexec/any_not_null_agg_tmpl.go, line 107 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/49934#-M9YwRco7F0kijsyBngM:-M9xaTht2nrXGPG9-kwE:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/432a1c53bdc486fe4474d45df045b9fd87ebfc53/pkg/sql/colexec/any_not_null_agg_tmpl.go#L107)):*
<details><summary><i>Previously, asubiotto (Alfonso Subiotto Marqués) wrote…</i></summary><blockquote>

:shrug: if it works. Yeah, I think the previous code called `SetOutputIndex` before aggregation computation.
</blockquote></details>

Done.
___
*[pkg/sql/colexec/ordered_aggregator.go, line 59 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/49934#-M9Yw_5P8n_N2DE6P-D3:-M9xaVXEB4LdPetYkqrO:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/16d5521b7cad659c977bf55a56bd600c0012122d/pkg/sql/colexec/ordered_aggregator.go#L59)):*
<details><summary><i>Previously, asubiotto (Alfonso Subiotto Marqués) wrote…</i></summary><blockquote>

I think it does because there are now two ways to do the same thing. I would remove it.
</blockquote></details>

Done.
